File: timetable-backend/app/models/course.py
from sqlalchemy import Column, Integer, String
from app.database import Base

class Course(Base):
    __tablename__ = "courses"

    id = Column(Integer, primary_key=True)
    name = Column(String, nullable=False)
    course_type = Column(String)  # theory / lab
    hours_per_week = Column(Integer)

File: timetable-backend-mern/models/Course.js
const mongoose = require('mongoose');

const courseSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    courseType: {
        type: String,
        enum: ['theory', 'lab'],
        required: true,
    },
    hoursPerWeek: {
        type: Number,
        required: true,
    },
}, {
    timestamps: true,
});

module.exports = mongoose.model('Course', courseSchema);
